modulus/models/gnn_layers/distributed_graph.py

```diff
@@ -27,6 +27,197 @@
     indexed_all_to_all_v,
 )
 
+@dataclass
+class GraphPartition:
+    partition_size: int
+    partition_rank: int
+    device: torch.device
+    # data structures for local graph
+    # of this current partition rank
+    local_offsets: torch.Tensor = None
```

Comment:
Should this be `Optional[torch.Tensor]` if default is `None`?

Reply:
I think generally, the defaults for many of this dataclass elements don't match the type annotations. This would cause lots of issues once we add static type checking https://github.com/NVIDIA/modulus/issues/121

Reply:
I am actually not sure about this dataclass here.
What I want here is
- a container for the related variables defining a graph partition
- without the need to specify all of these values when initializing the object
- hence, a dataclass with mostly defaults

For tensors, you can't really define a proper default here as things like size, device, and especially dtype are only known at a later stage. I'll go with Optional for now, so technically this is not really accurate